python/llm/src/bigdl/llm/transformers/modelling_bigdl.py

```diff
@@ -107,42 +109,53 @@ def from_pretrained(cls,
 
         :return: a model instance
         """
-        if native:
-            invalidInputError(dtype.lower() in ['int4', 'int8'],
-                              "Now we only support int4 and int8 as date type for weight")
-            ggml_model_path = pretrained_model_name_or_path
-            return cls.GGML_Model(model_path=ggml_model_path,
-                                  **kwargs)
-        else:
-            return cls.HF_Class.from_pretrained(pretrained_model_name_or_path,
-                                                *args, **kwargs)
+        try:
+            module = importlib.import_module(cls.GGML_Module)
+            class_ = getattr(module, cls.GGML_Model)
+            if native:
+                invalidInputError(dtype.lower() in ['int4', 'int8'],
+                                  "Now we only support int4 and int8 as date type for weight")
+                ggml_model_path = pretrained_model_name_or_path
+                model = class_(model_path=ggml_model_path, **kwargs)
+            else:
+                model = cls.HF_Class.from_pretrained(pretrained_model_name_or_path,
+                                                     *args, **kwargs)
+        except Exception as e:
+            invalidInputError(
+                False,
+                f"Could not load model from path: {pretrained_model_name_or_path}. "
+                f"Please make sure the CausalLM class matches "
+                "the model you want to load."
+                f"Received error {e}"
+            )
+        return model
 
 
 class LlamaForCausalLM(_BaseGGMLClass):
-    from bigdl.llm.ggml.model.llama import Llama
-    GGML_Model = Llama
+    GGML_Module = "bigdl.llm.models"
+    GGML_Model = "Llama"
     HF_Class = AutoModelForCausalLM
 
 
 class ChatGLMForCausalLM(_BaseGGMLClass):
-    from bigdl.llm.ggml.model.chatglm import ChatGLM
-    GGML_Model = ChatGLM
+    GGML_Module = "bigdl.llm.ggml.model.chatglm"
```

Comment:
why this path is different from other models?

Reply:
see https://github.com/intel-analytics/BigDL/blob/cbe0faccaceac35893e20cae9a3a5c761e960f28/python/llm/src/bigdl/llm/models.py#L26, the `chatglm` binary file for server is not ready yet.

Reply:
then don't forget to change it after it is ready.